core/Command/Encryption/MigrateKeyStorage.php

```diff
@@ -0,0 +1,276 @@
+	}
+
+	/**
+	 * move keys to new key storage root
+	 *
+	 * @param string $root
+	 * @param OutputInterface $output
+	 * @return bool
+	 * @throws \Exception
+	 */
+	protected function updateKeys(string $root, OutputInterface $output) {
+		$output->writeln("Start to update the keys:");
+
+		$this->updateSystemKeys($root);
+		$this->updateUsersKeys($root, $output);
+		$this->config->deleteSystemValue('encryption.key_storage_migrated');
+		return true;
+	}
+
+	/**
+	 * move system key folder
+	 *
+	 * @param string $root
+	 */
+	protected function updateSystemKeys($root) {
+		if (!$this->rootView->is_dir($root . '/files_encryption')) {
+			return;
+		}
+
+		$this->traverseKeys($root . '/files_encryption', null);
+	}
+
+	private function traverseKeys(string $folder, ?string $uid) {
+		$listing = $this->rootView->getDirectoryContent($folder);
+
+		foreach ($listing as $node) {
+			if ($node['mimetype'] === 'httpd/unix-directory') {
+				//ignore
+			} else {
+				$endsWith = function ($haystack, $needle) {
+					$length = strlen($needle);
+					if ($length === 0) {
+						return true;
+					}
+
+					return (substr($haystack, -$length) === $needle);
+				};
+
+				if ($node['name'] === 'fileKey' ||
+					$endsWith($node['name'], '.privateKey') ||
+					$endsWith($node['name'], '.publicKey') ||
+					$endsWith($node['name'], '.shareKey')) {
+					$path = $folder . '/' . $node['name'];
+
+					$content = $this->rootView->file_get_contents($path);
+
+					try {
+						$this->crypto->decrypt($content);
+						continue;
+					} catch (\Exception $e) {
+						// Ignore we now update the data.
+					}
+
+					$data = [
+						'key' => base64_encode($content),
+						'uid' => $uid,
+					];
+
+					$enc = $this->crypto->encrypt(json_encode($data));
+					$this->rootView->file_put_contents($path, $enc);
+				}
+			}
+		}
+	}
+
+	private function traverseFileKeys(string $folder) {
+		$listing = $this->rootView->getDirectoryContent($folder);
+
+		foreach ($listing as $node) {
+			if ($node['mimetype'] === 'httpd/unix-directory') {
+				$this->traverseFileKeys($folder . '/' . $node['name']);
+			} else {
+				$endsWith = function ($haystack, $needle) {
+					$length = strlen($needle);
+					if ($length === 0) {
+						return true;
+					}
+
+					return (substr($haystack, -$length) === $needle);
+				};
+
+				if ($node['name'] === 'fileKey' ||
+					$endsWith($node['name'], '.privateKey') ||
+					$endsWith($node['name'], '.publicKey') ||
+					$endsWith($node['name'], '.shareKey')) {
+					$path = $folder . '/' . $node['name'];
+
+					$content = $this->rootView->file_get_contents($path);
+
+					try {
+						$this->crypto->decrypt($content);
+						continue;
+					} catch (\Exception $e) {
+						// Ignore we now update the data.
+					}
+
+					$data = [
+						'key' => base64_encode($content)
+					];
+
+					$enc = $this->crypto->encrypt(json_encode($data));
+
+					$this->rootView->file_put_contents($path, json_encode($data));
```

Comment:
```suggestion
					$this->rootView->file_put_contents($path, $enc);
```

Otherwise we are writing back the old data.